python/dgl/nn/pytorch/conv/gatedgcnconv.py

```diff
@@ -0,0 +1,173 @@
+    >>> gatedGCN = GatedGCNConv(20, 12, 20)
+    >>> new_node_feats, new_edge_feats = gatedGCN(graph, node_feats, edge_feats)
+    >>> new_node_feats.shape, new_edge_feats.shape
+    (torch.Size([8, 20]), torch.Size([30, 20]))
+
+    """
+
+    def __init__(
+        self,
+        input_feats,
+        edge_feats,
+        output_feats,
+        dropout=0,
+        batch_norm=True,
+        residual=True,
+        activation=F.relu,
+    ):
+        super(GatedGCNConv, self).__init__()
+        self.dropout = nn.Dropout(dropout)
+        self.batch_norm = batch_norm
+        self.residual = residual
+
+        if input_feats != output_feats or edge_feats != output_feats:
+            self.residual = False
+
+        # Linearly transform the node features.
+        self.A = nn.Linear(input_feats, output_feats, bias=True)
+        self.B = nn.Linear(input_feats, output_feats, bias=True)
+        self.D = nn.Linear(input_feats, output_feats, bias=True)
+        self.E = nn.Linear(input_feats, output_feats, bias=True)
+
+        # Linearly transform the edge features.
+        self.C = nn.Linear(edge_feats, output_feats, bias=True)
+
+        # Batch normalization on the node/edge features.
+        self.bn_node = nn.BatchNorm1d(output_feats)
+        self.bn_edge = nn.BatchNorm1d(output_feats)
+
+        self.activation = activation
+
+    def forward(self, graph, feat, edge_feat):
+        """
+
+        Description
+        -----------
+        Compute gated graph convolution layer.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            The graph.
+        feat : torch.Tensor
+            The input feature of shape :math:`(N, D_{in})` where :math:`N`
+            is the number of nodes of the graph and :math:`D_{in}` is the
+            input feature size.
+        edge_feat : torch.Tensor
+            The input edge feature of shape :math:`(E, D_{edge})`,
+            where :math:`E` is the number of edges and :math:`D_{edge}`
+            is the size of the edge features.
+
+        Returns
+        -------
+        torch.Tensor
+            The output node feature of shape :math:`(N, D_{out})` where :math:`D_{out}`
+            is the output feature size.
+        torch.Tensor
+            The output edge feature of shape :math:`(E, D_{out})` where :math:`D_{out}`
+            is the output feature size.
+        """
+        with graph.local_scope():
+            # For residual connection
+            h_in = feat
+            e_in = edge_feat
+
+            graph.ndata["Ah"] = self.A(feat)
+            graph.ndata["Bh"] = self.B(feat)
+            graph.ndata["Dh"] = self.D(feat)
+            graph.ndata["Eh"] = self.E(feat)
+            graph.edata["Ce"] = self.C(edge_feat)
+
+            graph.apply_edges(fn.u_add_v("Dh", "Eh", "DEh"))
+
+            # Get edge feature
+            graph.edata["e"] = graph.edata["DEh"] + graph.edata["Ce"]
+            graph.edata["sigma"] = torch.sigmoid(graph.edata["e"])
```

Comment:
From the paper, the following operations are missing between L140 and 141 in the paper:

- batch normalization
- ReLU
- residual connection